hangman.py

```diff
@@ -0,0 +1,270 @@
+    elif letter not in set(secret_word):
+        tries = vowel_check(letter, tries)
+        print("Oops! That letter is not in my word:", check)
+    letters_guessed.add(letter)
+    l = [tries, warn, secret_word, letters_guessed, letter, check]
+    return l
+
+
+def hangman(secret_word, hints = False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    tries, warn = 6, 3
+    # List "letters_guessed" is used as a container for letters which have already been guessed
+    letters_guessed = set()
+    # Variabe "check" shows the length of the secret word and which letters have already been guessed
+    check = get_guessed_word(secret_word, letters_guessed)
+    while tries != 0 and is_word_guessed(secret_word, check):
+        print("-" * 13, "\nYou have", tries, "guesses left.")
+        print("Available letters:", get_available_letters(letters_guessed))
+        letter = str(input("Please guess a letter: ")).lower()
+
+        if len(letter) == 1:
+            l = check_element(tries, warn, letters_guessed, check, letter, hints)
+            tries, warn, secret_word, letters_guessed, letter, check = l[0], l[1], l[2], l[3], l[4], l[5]
+
+        # If user enters more or less than one symbol, the program issues a warning
+        else:
+            warn, tries = count(warn, tries)[0], count(warn, tries)[1]
+            print("Oops! Don't enter more or less than one symbol. You have", warn, "warnings left:", check)
+
+    if tries == 0:
+        print("-" * 13, '\nSorry, you ran out of guesses. The word was "' + secret_word + '".')
+    else:
+        print("-" * 13, "\nCongratulations, you won! Your total score for this game is:", tries * len(set(secret_word)))
+    return
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    my_word = my_word.replace(" ", "")
```

Comment:
In most of the times the `my_word` is the same, which means that you perform this replacement for that word many thousands times